com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicCurvesRenderer.cs

```diff
@@ -0,0 +1,252 @@
+using Unity.Burst;
+
+namespace Scripts.Importer
+{
+    [RequireComponent(typeof(AlembicCurves))]
+    [RequireComponent(typeof(MeshRenderer))]
+    [RequireComponent(typeof(MeshFilter))]
+    [ExecuteInEditMode]
+    public class AlembicCurvesRenderer : MonoBehaviour
+    {
+        AlembicCurves curves;
+        Mesh mesh;
+        [SerializeField] RenderMethod renderMethod;
+        [NonSerialized] RenderMethod prevRenderMethod;
+        ProfilerMarker setMeshProperties = new ProfilerMarker("SetMeshProperties");
+
+
+        enum RenderMethod
+        {
+            Line,
+            Strip
+        }
+
+        void OnEnable()
+        {
+            curves = GetComponent<AlembicCurves>();
+            //curves.OnUpdate += UpdateMesh;
+
+            mesh = new Mesh {hideFlags = HideFlags.DontSave};
+            GetComponent<MeshFilter>().sharedMesh = mesh;
+            var meshRenderer = GetComponent<MeshRenderer>();
+            if (meshRenderer.sharedMaterial == null)
+            {
+                meshRenderer.sharedMaterial = GetDefaultMaterial();
+            }
+            UpdateMesh(curves);
+        }
+
+        void UpdateMesh(AlembicCurves curves)
+        {
+            if (prevRenderMethod != renderMethod)
+            {
+                mesh.Clear();
+            }
+
+            prevRenderMethod = renderMethod;
+
+            switch (renderMethod)
+            {
+                case RenderMethod.Line:
+                    GenerateLineMesh(mesh, curves.Positions, curves.CurvePointCount);
+                    break;
+                case RenderMethod.Strip:
+                    //GeneratePlaneMesh(mesh, new[] {new Vector3(0, 0, 0), new Vector3(0, 1, 0), new Vector3(0, 2, 0)}, new[] {3}, new[] {0.1f, 0.75f, 0.1f});
+                    GeneratePlaneMesh(mesh, curves.Positions, curves.CurvePointCount, curves.Widths);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        void LateUpdate()
+        {
+            UpdateMesh(curves);
+        }
+
+        void OnDisable()
+        {
+            //curves.OnUpdate -= UpdateMesh;
+            DestroyImmediate(mesh);
+        }
+
+        void GeneratePlaneMesh(Mesh theMesh, IReadOnlyList<Vector3> positions, IReadOnlyList<int> curveCounts, IReadOnlyList<float> widths)
+        {
+            var curveCount = curveCounts.Count;
+
+            var indexArraySize = (positions.Count - curveCount) * 4;
+
+            using (var vertices = new NativeArray<Vector3>(positions.Count * 2, Allocator.Temp))
+            using (var indices = new NativeArray<int>(indexArraySize, Allocator.Temp))
+            {
+                unsafe
+                {
+                    var vertexPtr = (Vector3*)vertices.GetUnsafePtr();
+                    var indicesPtr = (int*)indices.GetUnsafePtr();
+
+                    var strandParticleBegin = 0;
+
+                    var dir = new Vector3(1, 0, 0);
```

Comment:
Shouldn't you try to compute a tangent for the curve instead of using the X axis? Right now on a moving character the mesh might look real ugly. You have the normals so you can compute a tangent at each step by getting the previous point to compute the curve segment direction.